src/Dapperer/Repository.Async.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Dapper;
+
+namespace Dapperer
+{
+    public abstract partial class Repository<TEntity, TPrimaryKey>
+        where TEntity : class, IIdentifier<TPrimaryKey>, new()
+    {
+        public virtual async Task<TEntity> GetSingleOrDefaultAsync(TPrimaryKey primaryKey)
+        {
+            string sql = _queryBuilder.GetByPrimaryKeyQuery<TEntity>();
+
+            using (IDbConnection connection = CreateConnection())
+            {
+                return (await connection.QueryAsync<TEntity>(sql, new { Key = primaryKey }).ConfigureAwait(false)).SingleOrDefault();
+            }
+        }
+
+        public virtual async Task<IList<TEntity>> GetByKeysAsync(IEnumerable<TPrimaryKey> primaryKeys)
+        {
+            string sql = _queryBuilder.GetByPrimaryKeysQuery<TEntity>();
+
+            using (IDbConnection connection = CreateConnection())
+            {
+                return (await connection.QueryAsync<TEntity>(sql, new { Keys = primaryKeys }).ConfigureAwait(false)).ToList();
+            }
+        }
+
+        public virtual async Task<IList<TEntity>> GetAllAsync()
+        {
+            string sql = _queryBuilder.GetAll<TEntity>();
+
+            using (IDbConnection connection = CreateConnection())
+            {
+                return (await connection.QueryAsync<TEntity>(sql).ConfigureAwait(false)).ToList();
+            }
+        }
+
+        public virtual async Task<Page<TEntity>> PageAsync(int skip, int take)
+        {
+            return await PageAsync(skip, take, null).ConfigureAwait(false);
+        }
+
+        public virtual async Task<TEntity> CreateAsync(TEntity entity)
+        {
+            string sql = _queryBuilder.InsertQuery<TEntity, TPrimaryKey>();
+
+            using (IDbConnection connection = CreateConnection())
+            {
+                if (_queryBuilder.GetBaseTableInfo<TEntity>().AutoIncrement)
+                {
+                    TPrimaryKey identity = (await connection.QueryAsync<TPrimaryKey>(sql, entity).ConfigureAwait(false)).SingleOrDefault();
+                    entity.SetIdentity(identity);
+                }
+                else
+                {
+                    await connection.ExecuteAsync(sql, entity).ConfigureAwait(false);
+                }
+
+                return entity;
+            }
+        }
+
+        public virtual async Task<int> CreateAsync(IEnumerable<TEntity> entities)
+        {
+            string sql = _queryBuilder.InsertQuery<TEntity, TPrimaryKey>(true);
+
+            using (IDbConnection connection = CreateConnection())
+            {
+                connection.Open();
```

Comment:
Well spotted, the create sync method had this for some wired reason. We don't really need it. I'lll remove it.